internal/buildpack/locator_type.go

```diff
@@ -74,8 +81,14 @@ func GetLocatorType(locator string, buildpacksFromBuilder []dist.BuildpackInfo)
 		return IDLocator, nil
 	}
 
-	if _, err := name.ParseReference(locator); err == nil {
-		return PackageLocator, nil
+	if strings.ContainsAny(locator, "@sha") || strings.Count(locator, "/") > 1 {
```

Comment:
We should add some tests for these changes.

Reply:
whoops, forgot to add them to the commit. The existing tests mostly cover it